lib/migrations/migrate/table-creator.js

```diff
@@ -27,7 +27,11 @@ function ensureTable(tableName, schemaName, trxOrKnex) {
     .then((data) => {
       return (
         !data.length &&
-        trxOrKnex.into(lockTableWithSchema).insert({ is_locked: 0 })
+        trxOrKnex.from(trxOrKnex.raw("?? (??)", [lockTableWithSchema, "is_locked"])).insert(function () {
+            return this.select(trxOrKnex.raw("?", [0])).whereNotExists(function () {
+                return this.select("*").from(lockTableWithSchema);
+            });
+        })
```

Comment:
While I believe this achieves what it aims to achieve, it's a little hard to mentally parse what's going on here. Should we extract it into a separate function, just so we can keep this bigger .then chain block a little easier to understand?

Reply:
👍 Makes sense. I extracted the logic into a helper function and updated the PR. Let me know what you think.